libscript/src/script-builder.cpp

```diff
@@ -1223,16 +1223,18 @@ void MCScriptAddVariableToHandlerInModule(MCScriptModuleBuilderRef self, MCNameR
     MCScriptHandlerDefinition *t_handler;
     t_handler = static_cast<MCScriptHandlerDefinition *>(self -> module . definitions[self -> current_handler]);
     
-    if (!MCMemoryResizeArray(t_handler -> local_count + 1, t_handler -> locals, t_handler -> local_count))
+    if (!MCMemoryResizeArray(t_handler -> local_type_count + 1, t_handler -> local_types, t_handler -> local_type_count) ||
```

Comment:
That wasn't going to work so I've updated by using an __extend_array function which adds an extra slot at the end of the array and returns its index.